src/app/locationForm/common/Form.jsx

```diff
@@ -15,8 +15,18 @@ class Form extends Component {
     }
   }
 
+  isEditing(props){
+    return props.value === null ||  
+      ( typeof props.value === 'object' ? 
+          Object.keys(props.value).some( key => !props.value[key] ) :
```

Comment:
IIUC, we will go into edit mode if even _one_ of the fields is empty.
Seeing as some fields may be optional and never receive a value (e,g. extension), I think it should be the other way around: we go into edit mode only if _none_ of the fields have a value.
We could verify with UX.

Reply:
Fixed in a463b359a697bd181e1ff8fedf18d39e65afc03a